docs/manuals/gui/viselements/charts/heatmap.md

```diff
@@ -1,261 +1,412 @@
-                            [53.0, 83.3, 72.7, 53.0],
-                        ],
-        'cities': ['New York', 'Los Angeles', 'Chicago', 'Houston']
-    }),
-    pd.DataFrame({
-        'seasons': ['Winter', 'Summer', 'Spring', 'Autumn'],
-    })
-]
+options = { "colorscale": "Portland" }
+```
+
+And reference that dictionary in the [*options*](../../chart/#p-options) property of the
+control:
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <|{data}|chart|type=heatmap|z=Temperatures|x=Seasons|y=Cities|options={options}|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:chart type="heatmap" z="Temperatures" x="Seasons" y="Cities" options="options">{data}</taipy:chart>
+        ```
+<figure>
+    <img src="../heatmap_colorscale-d.png" class="visible-dark" />
+    <img src="../heatmap_colorscale-l.png" class="visible-light" />
+    <figcaption>Specific Color Scale</figcaption>
+</figure>
 
-options = {'colorscale': 'Portland'}
+### Annotated Heatmap {data-source="gui:doc/examples/charts/heatmap-annotated"}
+
+You may want to display the actual value that is represented, at the appropriate location in the heatmap.
 
+We shall reuse the first example of this section, displaying the temperature for a given city
+in a given season in the appropriate heatmap cell.
+
+Here is the code that is needed:
+
+```py linenums="1"
 layout = {
-    'annotations': [],
-    'xaxis': {
-        'ticks': '',
-        'side': 'top'
+    "annotations": [],
+    "xaxis": {
+        "ticks": "",
+        "side": "top"
     },
-    'yaxis': {
-        'ticks': '',
-        'ticksuffix': ' ',
-        'width': 700,
-        'height': 700,
-        'autosize': False
+    "yaxis": {
+        "ticks": "",
+        "ticksuffix": " "
     }
 }
 
-for i in range(len(cities)):
-  for j in range(len(seasons)):
-    currentValue = temperature[i][j]
-    if (currentValue < 40 or currentValue > 70):
-        textColor = 'white'
-    else:
-        textColor = 'black'
-    result = {
-        'xref': 'x1',
-        'yref': 'y1',
-        'x': seasons[j],
-        'y': cities[i],
-        'text': str(temperature[i][j]) + '°F',
-        'font': {
-            'family': 'Arial',
-            'size': 12,
-            'color': textColor
-        },
-        'showarrow': False
+seasons = data["Seasons"]
+cities = data["Cities"]
+for city in range(len(cities)):
+  for season in range(len(seasons)):
+    temperature = data["Temperatures"][city][season]
+    annotation = {
+      "x": seasons[season],
+      "y": cities[city],
+      "text": f"{temperature}\N{DEGREE SIGN}C",
+      "font": {
+        "color": "white" if temperature < 9 or temperature > 26 else "black"
+      },
+      "showarrow": False
     }
-    layout['annotations'].append(result)
+    layout["annotations"].append(annotation)
 ```
+
+In lines 1 to 11, we create the object that is used in the [*layout*](../../chart/#p-layout)
```

Comment:
Just my thought: is it better if we put these explanations in code comments? Because there is no line number on the code block, the user will have to count.

Reply:
Line numbers are there. In the output doc.

Reply:
Oh, sorry, I can't see it on the live taipy doc, so I think here is the same